core/exception/hierarchy_spec.rb

```diff
@@ -0,0 +1,69 @@
+require_relative '../../spec_helper'
+
+describe "Exception" do
+  it "has the right class hierarchy" do
+    def save_pairs(parent, hash, pairs)
+      hash.each do |key, value|
+        pairs.push [parent, key]
+        if value.is_a?(Hash)
+          save_pairs(key, value, pairs)
+        end
+      end
+    end
+    hierarchy = {
+        Exception => {
+            NoMemoryError => nil,
+            ScriptError => {
+                LoadError => nil,
+                NotImplementedError => nil,
+                SyntaxError => nil,
+            },
+            SecurityError => nil,
+            SignalException => {
+                Interrupt => nil,
+            },
+            StandardError => {
+                ArgumentError => {
+                    UncaughtThrowError => nil,
+                },
+                EncodingError => nil,
+                FiberError => nil,
+                IOError => {
+                    EOFError => nil,
+                },
+                IndexError => {
+                    KeyError => nil,
+                    StopIteration => {
+                        ClosedQueueError => nil,
+                    },
+                },
+                LocalJumpError => nil,
+                NameError => {
+                    NoMethodError => nil,
+                },
+                RangeError => {
+                    FloatDomainError => nil,
+                },
+                RegexpError => nil,
+                RuntimeError => nil,
+                SystemCallError => nil,
+            ThreadError => nil,
```

Comment:
The indentation went wrong here and below

Reply:
Oops.  Adjusted the setting in RubyMine.